### PR TITLE
156 user can select preference for maximum distance of userprofiles (blackcards) presented to him

### PR DESCRIPTION
**Pull request by JoMuelle**
-Please merge at the same time as 201 (Backend for passing the maxRange)
There are parameters distance and maxRange as distance referes to the value of the rangesliders position and maxRange to the assigned range value of this position

### CODE REVIEW

**Comment by daveking7** [comment]
image
The only hint that those are my preferences is the edit preference button. can you change that?

File: src/components/views/BlackCardSelection.js [hunk 1]
```diff
@@ -81,10 +81,13 @@ const BlackCardSelection = () => {
...
             }
             catch (error) {
-                if(!(error.response.status === 404)){
+                if(error.response.status === 404){
+                    setBlackCard(null);
                     // && error.response.data.message === ""
+                }else{
                     console.error("Details:", error);
                     alert("Invalid Input:\n " + handleError(error));
                 }
```
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/components/views/BlackCardSelection.js:93]: It is already merged, but as there are no merge conflicts it doesnt display i guess. But ill edit a comment and do another push, maybe then itll show
image

[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/components/views/BlackCardSelection.js:93]: ahh just realised this is the client and not server ^^

File: src/components/views/EditUserPreferencePage.js [hunk 2]
```diff
@@ -67,20 +69,75 @@ const EditUserPreferencePage = () =>{
...
+    const scale = (distance) => {
+        const previousMarkIndex = Math.floor(distance / 9);
+        const previousMark = distanceMarks[previousMarkIndex];
+        const remainder = distance % 9;
+        if (remainder === 0) {
+            return previousMark.scaledValue;
+        }
+        const nextMark = distanceMarks[previousMarkIndex + 1];
+        const increment = (nextMark.scaledValue - previousMark.scaledValue) /9;
+        return remainder * increment + previousMark.scaledValue;
```
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/components/views/EditUserPreferencePage.js:130]: yes, its a bit fidgety but can be done, shall i do 1 increments up to 20 and then 10 increments up to 100?
But on the other hand i also think it's probably fine as it is, as i dont think they will care about the increment size here too much

File: src/components/views/BlackCardSelection.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/components/views/BlackCardSelection.js:93]: done now

File: src/components/views/EditUserPreferencePage.js [hunk 3]
```diff
@@ -67,20 +69,75 @@ const EditUserPreferencePage = () =>{
...
     };
 
+    function numFormatter(num) {
+        if(num > 1000){
+            return Math.floor((num/1000))*1000 + "km"
+        }
+        return num.toFixed(0) + "km"; // if value < 1000, nothing to do
```
[the remark this answers is not in the document]
Comment (JoMuelle) [reply src/components/views/EditUserPreferencePage.js:137]: done

**Pull request merged**